fabric/dataclient.go

```diff
@@ -585,6 +596,178 @@ func createRejectRoute(rid, path, host string, privs []interface{}) *eskip.Route
 	}
 }
 
+func createEmployeeAccessRoute(m *FabricMethod, eskipBackend *eskipBackend, allowedOrigins, userList []interface{}, accessType, name, namespace, host, path string) *eskip.Route {
+	r := &eskip.Route{
+		Id:     createRouteID("fg_eaccess", name, namespace, host, path, m.Method),
+		Path:   path,
+		Method: strings.ToUpper(m.Method),
+		Predicates: []*eskip.Predicate{
+			{
+				Name: predicates.HostAnyName,
+				Args: []interface{}{
+					host,
+					host + ":443",
+				},
+			},
+			{
+				Name: predicates.HeaderName,
+				Args: []interface{}{
+					"X-Forwarded-Proto",
+					"https",
+				},
+			},
+			{
+				Name: predicates.WeightName,
+				Args: []interface{}{
+					4, // TODO(sszuecs) needs checking
+				},
+			},
+		},
+		Filters: []*eskip.Filter{
+			{
+				// oauthTokeninfoAnyKV(realm", "/employees")
+				Name: filters.OAuthTokeninfoAnyKVName,
+				Args: []interface{}{
+					// TODO(sszuecs): should be configurable
+					"realm",
+					"/employees",
+				},
+			},
+			{
+				// oauthTokeninfoAllScope("uid")
+				Name: filters.OAuthTokeninfoAllScopeName,
+				Args: []interface{}{
+					// TODO(sszuecs): in the future should be configurable
+					"uid",
+				},
+			},
+			{
+				// unverifiedAuditLog("sub")
+				Name: filters.UnverifiedAuditLogName,
+				Args: []interface{}{
+					// TODO(sszuecs): in the future should be configurable
+					"sub",
+				},
+			},
+		},
+		BackendType: eskipBackend.Type,
+		Backend:     eskipBackend.backend,
+		LBAlgorithm: eskipBackend.lbAlgorithm,
+		LBEndpoints: eskipBackend.lbEndpoints,
+	}
+
+	// add optional ratelimit only default ratelimit
+	if m.Ratelimit != nil {
```

Comment:
Following three sections seems to be a direct duplication of sections from `createServiceRoute`. Maybe we can somehow share these parts? WDYT?

Reply:
there is quite some overlap in the route creations. I would like to refactor when we are done with the features.
I hope we can de-duplicate more than that :)